File: app/src/main/java/mrkm4ntr/twitterclient/util/BitmapCache.java
package mrkm4ntr.twitterclient.util;

import android.graphics.Bitmap;
import android.util.LruCache;

public class BitmapCache {

    private static int cacheSize = 6 * 1024;

    private static LruCache<String, Bitmap> mCache = new LruCache<String, Bitmap>(cacheSize) {
        @Override
        protected int sizeOf(String key, Bitmap bitmap) {
            return bitmap.getRowBytes() * bitmap.getHeight() / 1024;
        }
    };

    public static Bitmap getImage(String key) {
        Bitmap bitmap = mCache.get(key);
        if (bitmap != null && bitmap.isRecycled()){
            return null;
        } else {
            return bitmap;
        }
    }

    public static void setImage(String key, Bitmap bitmap) {
        if (!hasImage(key) && bitmap != null) {
            mCache.put(key, bitmap);
        }
    }

    public static boolean hasImage(String key) {
        if (getImage(key) == null) {
            return false;
        } else {
            return true;
        }
    }

    public static void clear() {
        mCache.evictAll();
    }
}

File: app/src/main/kotlin/mrkm4ntr/twitterclient/util/BitmapCache.kt
package mrkm4ntr.twitterclient.util

import android.graphics.Bitmap
import android.util.LruCache

object BitmapCache {

    private val cacheSize = 6 * 1024

    private val mCache = object : LruCache<String, Bitmap>(cacheSize) {
        override fun sizeOf(key: String, bitmap: Bitmap): Int {
            return bitmap.rowBytes * bitmap.height / 1024
        }
    }

    fun getImage(key: String): Bitmap? {
        val bitmap = mCache.get(key)
        if (bitmap != null && bitmap.isRecycled) {
            return null
        } else {
            return bitmap
        }
    }

    fun setImage(key: String, bitmap: Bitmap?) {
        bitmap?.let {
            if (!hasImage(key)) {
                mCache.put(key, bitmap)
            }
        }
    }

    fun hasImage(key: String): Boolean {
        return getImage(key) != null
    }

    fun clear() {
        mCache.evictAll()
    }
}
